platform/ios/src/MGLMapView.mm

```diff
@@ -3401,6 +3401,8 @@ - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
             positioningRect = annotationView.frame;
             
             [annotationView.superview bringSubviewToFront:annotationView];
+            
+            [annotationView setSelected:YES animated:animated];
```

Comment:
The `animated` property was intended to determine whether the map view pans to the selected annotation with animation: #3249. I don't think MapKit uses it to determine whether other things like the annotation view's visual appearance should animate.